pkg/cloud/gcp/gcp.go

```diff
@@ -407,6 +409,49 @@ func (c *GcpCloud) StopGCPNode(nodeConfig models.NodeConfig, clusterName string)
 	return nil
 }
 
+// ListRegions returns a list of regions for the GcpCloud instance.
+func (c *GcpCloud) ListRegions() []string {
+	regionListCall := c.gcpClient.Regions.List(c.projectID)
+	regionList, err := regionListCall.Do()
+	if err != nil {
+		return nil
+	}
+	regions := []string{}
+	for _, region := range regionList.Items {
+		regions = append(regions, region.Name)
+	}
+	return regions
+}
+
+// ListZonesInRegion returns a list of zones in a specific region for a given project ID.
+func (c *GcpCloud) ListZonesInRegion(region string) ([]string, error) {
+	zoneListCall := c.gcpClient.Zones.List(c.projectID)
+	zoneList, err := zoneListCall.Do()
+	if err != nil {
+		return nil, err
+	}
+	zones := []string{}
+	for _, zone := range zoneList.Items {
+		if zone.Region == fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/regions/%s", c.projectID, region) {
```

Comment:
can the fmt string be made a constant

Reply:
done

Reply:
in general when I talk about adding a constant. I suggest to add it to the constants package, or to constants header section of the file, or to a constants file inside the same directoty, but not to a constant defined in the same function
maybe we can agree on which style to use for this

Reply:
k sorry. will fix it